xdsl/printer.py

```diff
@@ -666,6 +667,11 @@ def print_int_or_question(value: IntAttr | NoneAttr) -> None:
 
         if isinstance(attribute, OpaqueSyntaxAttribute):
             self.print(">")
+
+        if isinstance(attribute, NoneType):
+            self.print("none")
+            return
+
```

Comment:
this needs to be higher

Reply:
Yep, More specifically anywhere before TypeAttribute: TrypeAttribute and below cases are more generic ones that will be applied to any attribute not falling in the earlier special cases.

Reply:
NB to the team: this deserve a comment ;)

Reply:
thanks for the spot :)